### DISCUSSION TITLE
ollama embeddings add models error

### DISCUSSION THREAD

**Discussion by burnedsyn** [langgenius/dify#8972] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

when i try to add an embeddings model with ollama it report a 404 error with the url set to http://mydomain.com:11434/api/embed
but the ollama embeddings api is api/embeddings i can just set the domain part where can i correct this ?
see my com for more informations.

I have updated the dify app this morning so i'm in the last version 9.1 and  i don't know if it worked in prior versions of dify because i just added them this morning

**Comment by burnedsyn** [comment]
Traceback (most recent call last):
  File "/app/api/controllers/console/workspace/models.py", line 152, in post
    model_provider_service.save_model_credentials(
  File "/app/api/services/model_provider_service.py", line 250, in save_model_credentials
    provider_configuration.add_or_update_custom_model_credentials(
  File "/app/api/core/entities/provider_configuration.py", line 390, in add_or_update_custom_model_credentials
    provider_model_record, credentials = self.custom_model_credentials_validate(model_type, model, credentials)
  File "/app/api/core/entities/provider_configuration.py", line 370, in custom_model_credentials_validate
    credentials = model_provider_factory.model_credentials_validate(
  File "/app/api/core/model_runtime/model_providers/model_provider_factory.py", line 123, in model_credentials_validate
    model_instance.validate_credentials(model, filtered_credentials)
  File "/app/api/core/model_runtime/model_providers/ollama/text_embedding/text_embedding.py", line 130, in validate_credentials
    raise CredentialsValidateFailedError(f"An error occurred during credentials validation: {str(ex)}")
core.model_runtime.errors.validate.CredentialsValidateFailedError: An error occurred during credentials validation: 404 Client Error: Not Found for url: http://XXXXX:11434/api/embed

a post request to this address is 404
but ollama answer very well on this address for now.
and on the model for llm size no problem it adds model.

Curl respond with the vector array on http://XXXXXX:11434/api/embeddings

**Reply by liberalchang** [reply]
@burnedsyn I have also encountered the same problem, may I ask if you have solved it

**Reply by crazywoola** [reply]
Please upgrade the Ollama to latest version, `api/embed` this is the new endpoint, and they have deprecated the old ones `api/embedings`

**Reply by jiangtingdong** [reply]
i  use  ollama version is 0.5.13 , it is the latest version,but the  endpoint, is api/embedings
